src/ape_ethereum/provider.py

```diff
@@ -1375,14 +1383,35 @@ def stream_request(self, method: str, params: Iterable, iter_path: str = "result
         payload = {"jsonrpc": "2.0", "id": 1, "method": method, "params": params}
         results = ijson.sendable_list()
         coroutine = ijson.items_coro(results, iter_path)
-        resp = requests.post(uri, json=payload, stream=True)
+        resp = requests_lib.post(uri, json=payload, stream=True)
         resp.raise_for_status()
 
         for chunk in resp.iter_content(chunk_size=2**17):
             coroutine.send(chunk)
             yield from results
             del results[:]
 
+    def batch_requests(self, requests: list[dict]) -> Any:
+        if not (uri := self.http_uri):
+            raise ProviderError("This provider has no HTTP URI and is unable to batch requests.")
+
+        for idx, request in enumerate(requests):
+            if "jsonrpc" not in request:
+                request["jsonrpc"] = "2.0"
+            if "id" not in request:
+                request["id"] = idx + 1
+
+        response = requests_lib.post(uri, json=requests)
+        try:
+            response.raise_for_status()
+        except HTTPError as err:
+            raise ProviderError(str(err)) from err
+
+        return [
+            self._get_result_from_rpc_response(uri, r, raise_on_failure=False)
+            for r in response.json()
```

Comment:
i remember a bug where erigon returned results out of order, so to be safe we can sort by id